### PR TITLE
Remove the inactive users #3205

### PR DESCRIPTION
**Pull request by Ranjeet-Kalsi**
Targets #3205 
**Issue-** In the User Management module, there are all the users are showing (Active/Inactive) in the users' tab. We need to show only active users in the user's tab.

### Description
1. found the view file , route, controller in module -> user 
2. then found that there are a problem in UsersService.php Index function 
after some changes as follows 
       return User::with('roles')
            ->whereNull('deleted_at')
            ->orderBy('name')
            ->get();
    }

### Checklist:

- [x] I have performed a self-review of my own code.
- [x] I have also testing the another tabs.

active

### CODE REVIEW

**Review by Ayush8923 — CHANGES_REQUESTED** [review]
Left some comments. Please check.

File: Modules/User/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -21,7 +21,7 @@ public function __construct(UserServiceContract $service)
     public function index()
     {
         $this->authorize('viewAny', User::class);
-        $users = $this->service->index();
+        $users = User::has('roles')->with('roles')->get();
```
[the remark this answers is not in the document]
Comment (Ranjeet-Kalsi) [reply Modules/User/Http/Controllers/UserController.php:24]: > I don't think, this return only active user. Please check the other places like the Client module `Key account manager` dropdown. In this dropdown, only active users have been shown. Please take reference to the below screenshot image

@Ayush8923   I will check it

[the remark this answers is not in the document]
Comment (Ranjeet-Kalsi) [reply Modules/User/Http/Controllers/UserController.php]: @Ayush8923  I applied  the logic in service file 

but how can i testing the CRM module 
I have seen in my screen as per attached screenshots.
please suggest how can i applied in crm module
crm
inactive
active

File: Modules/User/Services/UserService.php [hunk 2]
```diff
@@ -14,7 +14,7 @@ class UserService implements UserServiceContract
      */
     public function index()
     {
-        return User::with('roles')
+        return User::has('roles')->with('roles')
```
Comment (Ayush8923) [inline Modules/User/Services/UserService.php:17]: As far as I can check, these changes have not removed the inactive users.

**Review by Ayush8923 — CHANGES_REQUESTED** [review]

File: Modules/User/Services/UserService.php (hunk 2 above)
Comment (Ranjeet-Kalsi) [reply Modules/User/Services/UserService.php:17]: > As far as I can check, these changes have not removed the inactive users. You need to check the logic in the Client Module, Key Account Manager dropdown. How are the only active users shown in there? Please let me know if there are any questions/ confusion regarding this.
 
@Ayush8923 Thanks for your suggestion 
i will check it out 
I have import the database of portal
If any problem is that then i Connect with you

Comment (Ranjeet-Kalsi) [reply Modules/User/Services/UserService.php:17]: @Ayush8923 you assigne me task https://github.com/ColoredCow/portal/issues/3205
**issue** -Remove the inactive users in the user's tab. #3205
In the User Management module, there are all the users are showing (Active/Inactive) in the users' tab. We need to show only active users in the user's tab.

I worked on it it works properly only active user shown 
now why should i waste my time in CRM module.
first you should review on task 3205 if the task completed 
Then assigned a new task with CRM module.

[the remark this answers is not in the document]
Comment (Ranjeet-Kalsi) [reply Modules/User/Services/UserService.php:17]: ok sorry for that i am confused With crm module now 

but ->whereNull('deleted_at')

[the remark this answers is not in the document]
Comment (Ayush8923) [reply Modules/User/Services/UserService.php:17]: So the scenario is that when the user becomes inactive then the date is stored in the `deleted_at` column in the user table and if the user is active then the `deleted_at` column in the user table has the corresponding null value of the user. So we fetch only those users from the database whose `deleted_at` is null. We can't tell from the roles that this user is active or inactive.
Please let me know if you have any other questions/ confusion. So we can connect.

[the remark this answers is not in the document]
Comment (Ranjeet-Kalsi) [reply Modules/User/Services/UserService.php:17]: ok @Ayush8923  now working thanks for your suggestion
in my database 30 entries without names they create confused me 
now active user displayed using your opinion ->whereNull('deleted_at')

**Pull request merged**